### PR TITLE
cli: Add bash and zsh autocomplete

### LINKED ISSUE #2094: mc should have auto-completion facility
## Expected behaviour

Like many common CLI utilities that are structured as a single program with numerous subcommands (e.g. git/docker), mc will benefit from auto-completion support, especially for the bash shell.

## mc version
- (paste output of `mc version`)

### PR DESCRIPTION
**Pull request by harshavardhana**
Fixes #2094

### CODE REVIEW

File: autocomplete/bash_autocomplete [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#! /bin/bash
```
Comment (vadmeste) [inline autocomplete/bash_autocomplete]: Unneeded space?

Comment (vadmeste) [inline autocomplete/bash_autocomplete]: between `"!` and `/bin/bash`

File: autocomplete/bash_autocomplete [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+_cli_bash_autocomplete() {
+     local cur opts base
+     COMPREPLY=()
+     cur="${COMP_WORDS[COMP_CWORD]}"
+     opts=$( ${COMP_WORDS[@]:0:$COMP_CWORD} --compgen )
+     COMPREPLY=( $(compgen -W "${opts}" -- ${cur}) )
+     return 0
+ }
+  
+ complete -F _cli_bash_autocomplete $PROG
```
Comment (vadmeste) [inline autocomplete/bash_autocomplete]: We probably need to add `-q` flag here to avoid printing update message when compgen is called.

Comment (harshavardhana) [inline autocomplete/bash_autocomplete]: where does it show that?

Comment (harshavardhana) [inline autocomplete/bash_autocomplete]:
```sh
$ PROG=mc source autocomplete/bash_autocomplete 
$ 
```

Don't see any message.

Comment (harshavardhana) [inline autocomplete/bash_autocomplete]: oh you mean the `mc update` message?

Comment (vadmeste) [inline autocomplete/bash_autocomplete]: > oh you mean the mc update message?

Besides, we used to have that banner which is shown for each command when mc binary becames outdated, we don't have that thing anymore?

Comment (harshavardhana) [inline autocomplete/bash_autocomplete]: Fixed that too..

File: autocomplete/bash_autocomplete (hunk 1 above)
Comment (harshavardhana) [inline autocomplete/bash_autocomplete]: Fixed

**Pull request merged**